controllers/vsl.go

```diff
@@ -19,61 +30,91 @@ func (r *VeleroReconciler) ValidateVolumeSnapshotLocations(log logr.Logger) (boo
-					r.Log.Info("VSL for AWS specified, but AWS plugin not present, might be a misconfiguration")
+			//in AWS, region is a required field
+			if len(vslSpec.Config[Region]) == 0 {
+				return false, errors.New("region for AWS VSL is not configured, please ensure a region is configured")
+			}
+			// check for invalid config key
+			for key := range vslSpec.Config {
+				if key != Region && key != Profile {
+					return false, errors.New("invalid AWS config value")
 				}
-
-				//TODO: Add warn/error messages to Velero CR status field
 			}
+			//checking the aws plugin, if not present, throw warning message
+			if !contains(velero.Spec.DefaultVeleroPlugins, AWSProvider) {
+				r.Log.Info("VSL for AWS specified, but AWS plugin not present, might be a misconfiguration")
+
+				r.EventRecorder.Event(&vsl,
+					corev1.EventTypeWarning,
+					"VolumeSnapshotLocation is invalid",
+					fmt.Sprintf("could not validate vsl for AWS plugin on: %s/%s", vsl.Namespace, vsl.Name),
+				)
+			}
+		}
 
-			//GCP
-			if vslSpec.Provider == "gcp" {
-
-				//validation for GCP
-				if len(vslSpec.Config["region"]) == 0 {
-					r.Log.Info("region for GCP VSL is not configured, please check if a region might be needed")
-				}
+		//GCP
+		if vslSpec.Provider == GCPProvider {
 
-				//checking the gcp plugin, if not present, throw warning message
-				if !contains(velero.Spec.DefaultVeleroPlugins, "gcp") {
-					r.Log.Info("VSL for GCP specified, but GCP plugin not present, might be a misconfiguration")
+			// no other required fields for gcp
+			// check for invalid config key
+			for key := range vslSpec.Config {
+				if key != SnapshotLocation && key != Project {
+					return false, errors.New("invalid GCP config value")
 				}
-
-				//TODO: Add warn/error messages to Velero CR status field
-
 			}
-
-			//Azure
-			if vslSpec.Provider == "azure" {
-
-			//validation for Azure
-			if len(vslSpec.Config["region"]) == 0 {
-				r.Log.Info("region for Azure VSL is not configured, please check if a region might be needed")
+			//checking the gcp plugin, if not present, throw warning message
+			if !contains(velero.Spec.DefaultVeleroPlugins, "gcp") {
+				r.Log.Info("VSL for GCP specified, but GCP plugin not present, might be a misconfiguration")
+
+				r.EventRecorder.Event(&vsl,
+					corev1.EventTypeWarning,
+					"VolumeSnapshotLocation is invalid",
+					fmt.Sprintf("could not validate vsl for GCP plugin on: %s/%s", vsl.Namespace, vsl.Name),
+				)
 			}
+		}
 
-				//checking the azure plugin, if not present, throw warning message
-				if !contains(velero.Spec.DefaultVeleroPlugins, "azure") {
-					r.Log.Info("VSL for Azure specified, but Azure plugin not present, might be a misconfiguration")
+		//Azure
+		if vslSpec.Provider == Azure {
+			// no other required fields for gcp
+			// check for invalid config key
+			for key := range vslSpec.Config {
+				if key != ApiTimeout && key != ResourceGroup &&
```

Comment:
Since this is only 4 keys the chained if conditions aren't terrible, but consider (in case more keys are added later)

```
validAzureKeys := []string{ApiTimeout, ResourceGroup, ...}

or

validAzureKeys := map[string]bool{
    ResourceGroup: true,
...
}
```
Then you can either use the `contains` func or just do `valid := validAzureKeys[key]` to see if its a valid key